Library/Homebrew/cmd/install.rb

```diff
@@ -120,6 +120,10 @@ class InstallCmd < AbstractCommand
           [:switch, "--overwrite", {
             description: "Delete files that already exist in the prefix while linking.",
           }],
+          [:switch, "--ask", {
+            description: "Ask for confirmation before downloading and installing software. " \
+                         "Print bottles and dependencies download size and install size.",
+          }],
```

Comment:
```suggestion
          [:switch, "--ask", {
            description: "Ask for confirmation before downloading and installing formulae. " \
                         "Print bottles and dependencies download size and install size.",
            env: :ask,
          }],
```
will allow you to just use `args.ask?` below.

- Do you plan on supporting this for e.g. `brew upgrade` or `brew reinstall`? If not, might want to do `:install_ask` instead.

Reply:
thanks for the update! it will be clearer. I could implement it in upgrade and reinstall but it will not be as relevant as some contents could be deleted. like the pacman package manager, I could put a `net install` feature to show the difference between the size of newly installed packages and the old ones. I don't know if that could be supported with brew.
I will put `install_ask` for now and open another issue to also implement it to `brew upgrade` and `brew reinstall`.

Reply:
If you think you'll open others: good to keep as `:ask` 👍🏻